### PR TITLE
Fixes for issue #1701

### LINKED ISSUE #1701: Missing data from carbonlink cache when request crosses retention boundaries
A request for a graph that starts before the lowest retention period has lines that end randomly, which looks bad.  Also data extraction for any automated data analysis ends in many Nones (in my case it's upwards of 20-40 minutes), which isn't good for data analysis.

Example:  Current time was 14:50.  Data is/was available in the caches.
image

This is a by-product of #1002.  The WhisperReader is explicitly not querying CarbonLink.  I believe this is NOT what we want, but can we gather a list of the conditions that we do need to handle?

### PR DESCRIPTION
**Pull request by cbowman0**
This fixes #1701 

* Expand the merge_with_cache to consolidate data using the aggregation method passed in.
* Always issue carbonlink request for WhisperReader

### CODE REVIEW

File: webapp/tests/test_readers.py [hunk 1]
```diff
@@ -5,6 +5,39 @@
...
+
+        values = readers.merge_with_cache(
+            cached_datapoints=cache_results,
+            start=start,
+            step=step,
+            values=values,
+            func='sum'
+        )
+
+        #Compute what this is.
```
Comment (obfuscurity) [inline webapp/tests/test_readers.py]: Minor nit, adding a space after the `#`.

Comment (cbowman0) [inline webapp/tests/test_readers.py]: Yep.  Agreed.

Comment (cbowman0) [inline webapp/tests/test_readers.py]: Actually, that comment needs help in clarity, too.  It was to remind me to add that code there, but that was done.

**Comment by cbowman0** [comment]
That timing issue in test_render bit again.  Maybe I'll track that down sometime.  Can someone with perms ask Travis CI to try again?

**Comment by obfuscurity** [comment]
Job restarted.

**Comment by obfuscurity** [comment]
👍

**Comment by deniszh** [comment]
👍

**Pull request merged**